### ISSUE TITLE
Get rgee to respect reticulate::use_condaenv() 

### ISSUE THREAD

**Issue by grantmcdermott** [csaybar/rgee#21]
I had some trouble connecting **rgee** to my existing Python EE API, which I had installed some time ago via conda. Luckily, the `ee_set_python_version(python_path = "CONDA_PATH")` solution suggested in #17 worked for me.

That being said, I do think the above workaround goes against existing practices for calling up conda environments via **reticulate**. For example, I would expect the following to work:

```r
library(rgee)
library(reticulate)

## Use previously created "ee" conda environment with all requisite Python modules 
use_condaenv("ee") 

## Test Python connection working (it is)
os <- import("os")
os$listdir(".")

## Try to initialize ee module
## Doesn't work b/c it ignores the reticulate::use_condaenv() call above and defaults to system Python
ee_Initialize()
```

**In short:** Would it be possible to get **rgee** to respect `reticulate::use_condaenv()`? 

Again, I think this makes sense given that:
1. It's how **reticulate** [expects](https://rstudio.github.io/reticulate/reference/use_python.html) to connect to conda environments.
2. Most existing EE API users will likely have installed through conda, since it's the [recommended method](https://developers.google.com/earth-engine/python_install-conda.html) for local installation.

Regardless, thanks for an exciting package!

**Comment by csaybar** [comment]
@grantmcdermott thanks for open this issue!

I would like to start mentioned that `ee_set_python_version` is not compulsory. And, the code that you attached should work perfectly.

The function  `ee_set_python_version` defines, into the .renviron file, the [system variables](https://rstudio.github.io/reticulate/articles/versions.html#providing-hints): **RETICULATE_PYTHON** and **RETICULATE_PYTHON_ENV**.

RETICULATE_PYTHON will force to reticulate to just considered the Python path, set on it. While  RETICULATE_PYTHON_ENV will do the same for Python virtuals environments.

In the next commit of `rgee`, I will change the install module to permit that `ee_set_python_version` changes also the python env. Currently, this looks like this:
https://github.com/csaybar/rgee/blob/5ce246a65ed8bb1ddbf734a5a00cf595520e7323/R/ee_install.R#L49
and this is a terrible idea, **thanks for the feedback!.**

If you are interested in continues using `ee_set_python_version`. Your example will need this structure to work:

```r
library(rgee)

# Find all Python environments in the system.
ee_discover_pyenvs()

## Use previously created "ee" conda environment with all requisite Python modules 
use_condaenv(python_path = '/home/user/.virtualenvs/ee/bin/python',env="ee") 

## Test Python connection working (it is)
os <- import("os")
os$listdir(".")

## Try to initialize ee module
## This should work now 
ee_Initialize()
```

Honestly, I do not sure if using systems variables is the most appropriate way to deal with the need of rgee for a Python connection, but definitely will be more friendly with users with no experience with reticulate and Python. However, the options `use_condaenv` and `use_virtualenv` will do not work. I will really appreciate ideas on how to create an installation framework that satisfies all users.

By the way, could you open another issue with the installation problem that you mentioned at the beginning of your comment?,  I really interested in solve problems when `ee_set_python_version` is not used. Thank you very much for your interest in rgee!

**Comment by csaybar** [comment]
The commit was made. We decide renamed `ee_set_python_version`  to `ee_set_pyenv`.
Please before retry clean your system variables as follow:

```r
library(rgee)
ee_clean_pyenv()
# Restart R to see changes
```

[Requirements](https://github.com/csaybar/rgee#requirements)

**Comment by grantmcdermott** [comment]
Hi @csaybar, thanks very much for the prompt and detailed reply.

I've tried your suggestion and it works... but it also has the very undesirable side-effect of completely re-writing your .Renviron file! 

So all existing environment variables and API keys are erased :'(

Luckily, I keep a back-up of my .Renviron, so I could experiment a bit to see where the problem occurs. The culprit is the new `ee_set_pyenv()` function. (I didn't check but I'm pretty sure the old `ee_set_python_version()` would have suffered from the same problem.)

Could you please amend the function to fix this overwriting problem? 

Thanks!

---

PS. For convenience, here's an example of how the the `tidycensus::census_api_key()` function does it. You can see the various safeguards, including the creation of a backup .Renvion file, etc.

```
## tidycensus::census_api_key
function (key, overwrite = FALSE, install = FALSE) 
{
    if (install) {
        home <- Sys.getenv("HOME")
        renv <- file.path(home, ".Renviron")
        if (file.exists(renv)) {
            file.copy(renv, file.path(home, ".Renviron_backup"))
        }
        if (!file.exists(renv)) {
            file.create(renv)
        }
        else {
            if (isTRUE(overwrite)) {
                message("Your original .Renviron will be backed up and stored in your R HOME directory if needed.")
                oldenv = read.table(renv, stringsAsFactors = FALSE)
                newenv <- oldenv[-grep("CENSUS_API_KEY", oldenv), 
                  ]
                write.table(newenv, renv, quote = FALSE, sep = "\n", 
                  col.names = FALSE, row.names = FALSE)
            }
            else {
                tv <- readLines(renv)
                if (any(grepl("CENSUS_API_KEY", tv))) {
                  stop("A CENSUS_API_KEY already exists. You can overwrite it with the argument overwrite=TRUE", 
                    call. = FALSE)
                }
            }
        }
        keyconcat <- paste0("CENSUS_API_KEY='", key, "'")
        write(keyconcat, renv, sep = "\n", append = TRUE)
        message("Your API key has been stored in your .Renviron and can be accessed by Sys.getenv(\"CENSUS_API_KEY\"). \nTo use now, restart R or run `readRenviron(\"~/.Renviron\")`")
        return(key)
    }
    else {
        message("To install your API key for use in future sessions, run this function with `install = TRUE`.")
        Sys.setenv(CENSUS_API_KEY = key)
    }
}
```

**Comment by csaybar** [comment]
This is so shameful ... I was also one of the victims and I didn't realize until I read your message :'( . I fix this bug in the last commit.  `ee_set_pyenv` now works quite similar to `census_api_key` and also generates a `.Renviron_backup` file. 

I going to close this issue but feel free to open it again if you consider it necessary.
Thanks a lot to help us to enhance `rgee`!

**Comment by grantmcdermott** [comment]
Super, thanks Cesar.

BTW, I've really been enjoying the package so far. The integration with sf is fantastic; being able to combine the power of EE operations and datasets with local sf objects has removed some major bottlenecks for me. I hope you'll consider submitting the package for a TWEEDS talk next year! https://tweeds.io/

**Comment by csaybar** [comment]
I am really happy to hear that :). We have some extra ideas that we will implement in the near future to facilitate, even more, the connection with Earth Engine. [tweeds](https://tweeds.io/) workshop looks fantastic I hope to have the opportunity to attend one day!.
